js/modules/k6/execution/execution.go

```diff
@@ -325,6 +325,10 @@ func (o *tagsDynamicObject) Get(key string) goja.Value {
 // In any other case, if the Throw option is set then an error is raised
 // otherwise just a Warning is written.
 func (o *tagsDynamicObject) Set(key string, val goja.Value) bool {
+	if goja.IsUndefined(val) || goja.IsNull(val) {
+		o.State.Tags.Set(key, "")
+		return true
+	}
 	switch val.ExportType().Kind() { //nolint:exhaustive
```

Comment:
I'm wondering if we should set the tags as the empty case that way :thinking: especially since we have a "strict mode" option  (`o.State.Options.Throw.Bool`)

Maybe we should use something like this :thinking:   
```suggestion
	kind := reflect.Invalid
	if val != nil {
		kind = val.ExportType().Kind()
	}

	switch kind { //nolint:exhaustive
```

WDYT?

Reply:
> I saw the logic we have if we do the same from the options I changed because it wouldn't be consistent

I'm not sure that I know what this logic is :disappointed:

Reply:
I mean the example reported in the issue's description:

```javascript
import exec from 'k6/execution';

export const options = {
	tags: {
		undef: undefined,
		nul: null,
	}
};

export default function () {
	console.log('' === exec.vu.tags.nul) // true
	console.log('' === exec.vu.tags.undef) // true
}
```

Reply:
Not really certain what the behaviour needs to be :(. It setting those to `''` seems bad. IMO it should either:
1. throw an exception as it just being pointless (maybe depending on `options.throw`)
2. actually set it to `null` and `undefined`. And return those on `Get` and (true on `Has`). But I am not really certain what that will be used for and ... it likely will need to stringify in an output to `"null"`/`"undefined"` or `''` all of which seem bad.
So I guess I am leaning towards `1` :shrug: 

I was going to argue this should delete a tag .. but `delete` exists so it seems like this isn't needed

Reply:
I would be inclined to go for @mstoykov's 1 too. It could probably throw a JS `TypeError`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError?

Reply:
@na-- do you want to do the same if we set from the exported options or just from the execution API?

Reply:
>It could probably throw a JS TypeError: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError?

@oleiade if `--throw` is set then we use `common.Throw` that accepts just a Go error type. `goja.NewTypeError` returns a `*goja.Object` so I guess we can't use it.

Reply:
You can just `panic` with the `TypeError` directly ?

Reply:
Fix for the panic, that panics? :see_no_evil:

Reply:
What do you think `common.Throw` does ?
https://github.com/grafana/k6/blob/27f2ead359dac24b4131dac4b57c95c058f06e48/js/common/util.go#L35-L41

Reply:
Damn, I also tried it. It's the habit of not using different types from err and string. :facepalm: